pyopencl/array.py

```diff
@@ -612,12 +612,9 @@ def __init__(
 
                 shape = (shape,)
 
-            size = 1
-            for dim in shape:
-                size *= dim
-                if dim < 0:
-                    raise ValueError(
-                        f"negative dimensions are not allowed: {shape}")
+            size = np.prod(shape)
+            if size < 0:
+                raise ValueError(f"negative dimensions are not allowed: {shape}")
```

Comment:
Technically, this error will now not be triggered if there is an even number of negative shape axes.